back/app/services/side_fx_custom_field_service.rb

```diff
@@ -19,6 +19,8 @@ def after_update(custom_field, current_user)
   end
 
   def before_destroy(custom_field, _current_user)
+    return unless custom_field.resource_type == 'User'
```

Comment:
A main bugfix (of 2).

Since `UserCustomFieldService.new.delete_custom_field_values` is only concerned with removing `User.custom_field_values` (both the key & value)), we now return a no-op unless `custom_field.resource_type == 'User'`